core/src/main/java/bio/terra/pearl/core/service/kit/pepper/StubPepperDSMClient.java

```diff
@@ -8,14 +8,19 @@
 import bio.terra.pearl.core.service.kit.pepper.PepperDSMClient;
 import bio.terra.pearl.core.service.kit.pepper.PepperKitAddress;
 import bio.terra.pearl.core.service.kit.pepper.PepperKitStatus;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.RandomUtils;
 import org.springframework.context.annotation.Lazy;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
```

Comment:
PepperDSMClientProvider is essentially just a switcher configuration class that switches which client gets used based on the value of USE_LIVE_DSM.  If true, the live client gets used, otherwise, the stub client gets used.

And yes, StubDSMClient is used for local testing where spinning up or connecting to Dev DSM isn't worth it.  It is currently too stubby for my liking.  Ideally, it would be more fully-functioning, and maybe using an internal cache of it's own.  This is why you're seeing it gradually accumulate semi-realistic behavior like in this PR, and gradually mimic more of Pepper's responses more precisely.